### PR TITLE
NIFI-4406 Updated ExecuteScript to use Searchable.

### PR DESCRIPTION
**Pull request by MikeThomsen**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by MikeThomsen** [comment]
@markap14 can you take a look? This is from a feature ticket you wrote up.

File: nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ExecuteScript.java [hunk 1]
```diff
@@ -83,7 +88,7 @@
 @Stateful(scopes = {Scope.LOCAL, Scope.CLUSTER},
         description = "Scripts can store and retrieve state using the State Management APIs. Consult the State Manager section of the Developer's Guide for more details.")
 @SeeAlso({InvokeScriptedProcessor.class})
-public class ExecuteScript extends AbstractSessionFactoryProcessor {
+public  class ExecuteScript extends AbstractSessionFactoryProcessor implements Searchable {
```
Comment (patricker) [inline nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ExecuteScript.java]: Is that 2 spaces between `public` and `class`?

Comment (MikeThomsen) [reply nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ExecuteScript.java]: Not anymore...

**Comment by markap14** [comment]
Hey @MikeThomsen thanks for submitting the PR. Sorry it's taken me so long to get around to reviewing it! I did a quick test and it does work correctly. However, I'm a bit concerned about the fact that it returns the entire script body if it matches. While this is the same behavior if the "Script Body" property is set, I feel like if a file is specified, there's a much higher likelihood that the script is quite large. What do you think about only returning the line in the script that matches? Or, rather, a match for each line in the script body that matches?

**Comment by MikeThomsen** [comment]
Wouldn't be a problem to refactor it to scan line by line.

**Comment by markap14** [comment]
Thanks. I think it would be helpful.

**Comment by MikeThomsen** [comment]
@markap14 rebased and pushed that change.

**Comment by markap14** [comment]
Thanks @MikeThomsen for the improvement! And thanks for your patience getting it merged in. Was able to test and verify that it behaves exactly as expected. +1 merged to master.

**Comment by MikeThomsen** [comment]
@markap14 does the search system read from controller services as well?

**Comment by markap14** [comment]
@MikeThomsen you mean does it search the configuration of Controller Services? If so, then yes, it does.

**Pull request closed without merging**